app/models/profile_field.rb

```diff
@@ -0,0 +1,12 @@
+class ProfileField < ApplicationRecord
+  # Key names follow the Rails form helpers
+  enum input_type: {
+    text_field: 0,
+    text_area: 1,
+    check_box: 2,
+    color_field: 3
+  }
+
+  validates :label, presence: true, uniqueness: { case_sensitive: false }
+  validates :active, presence: true
```

Comment:
@citizen428 

Since [presence uses .blank?](https://edgeguides.rubyonrails.org/active_record_validations.html#presence) and since false.blank? is true, if we want to validate the presence of a boolean field should we not be using:
```validates : active, inclusion: { in: [true, false] }```

otherwise, trying to set active to false results in the ffg rollback:
```
[22] pry(main)> p.update(active: false)
   (0.3ms)  BEGIN
  ProfileField Exists? (0.4ms)  SELECT 1 AS one FROM "profile_fields" WHERE LOWER("profile_fields"."label") = LOWER($1) AND "profile_fields"."id" != $2 LIMIT $3  [["label", "Name"], ["id", 1], ["LIMIT", 1]]
   (0.2ms)  ROLLBACK
=> false


p.update(active: 0)
   (0.3ms)  BEGIN
  ProfileField Exists? (0.4ms)  SELECT 1 AS one FROM "profile_fields" WHERE LOWER("profile_fields"."label") = LOWER($1) AND "profile_fields"."id" != $2 LIMIT $3  [["label", "Name"], ["id", 1], ["LIMIT", 1]]
   (0.2ms)  ROLLBACK
=> false
```

Reply:
nice catch!

Reply:
@Ridhwana You're right, it's also in the [documentation of the uniqueness validator](https://api.rubyonrails.org/classes/ActiveRecord/Validations/ClassMethods.html#method-i-validates_presence_of):

> If you want to validate the presence of a boolean field (where the real values are true and false), you will want to use `validates_inclusion_of :field_name, in: [true, false]`.

Not the first time this got me and probably won't be the last 😉